tests/sentry/integrations/test_client.py

```diff
@@ -142,6 +153,106 @@ def test_default_redirect_behaviour(self):
         resp = ApiClient().delete("http://example.com/3", allow_redirects=True)
         assert resp.status_code == destination_status
 
+    def test_ConnectionError_handling(self):
+        client = ApiClient()
+
+        with mock.patch.object(
+            client, "track_response_data", wraps=client.track_response_data
+        ) as track_response_data_spy:
+            with mock.patch(
+                "requests.sessions.Session.get",
+                side_effect=ConnectionError("foo"),
+            ):
+                with pytest.raises(ApiHostError):
```

Comment:
I think the three `with` can be chained in the same one if separated by commas. I can't remember (`with foo, bar, baz:`)

Reply:
Yeah, you definitely can. I tried doing it that way and it felt more visually muddled, so I opted for nesting. What I actually wanted was [this](https://docs.python.org/3/whatsnew/3.10.html#parenthesized-context-managers), but alas, we're only on 3.8.

Here's what it looks like if you combine them:
![image](https://user-images.githubusercontent.com/14812505/227029311-7da7c630-68c7-40c3-aa08-40f003a0af11.png)

The `pytest.raises` is all right, but the second `mock.patch` gets pretty lost in the shuffle, visually. I agree that the nesting is also not the nicest, but at least it's clear what the pieces are.